neuromaps/nulls/nulls.py

```diff
@@ -409,56 +428,99 @@ def _surf_surrogates(data, atlas, density, parcellation, distmat, n_proc):
 
 
 def _vol_surrogates(data, atlas, density, parcellation, distmat, **kwargs):
+
     if atlas != 'MNI152':
         raise ValueError('Cannot compute volumetric surrogates if atlas is '
-                         f'"MNI152". Received: {atlas}')
+                         'not 'f'"MNI152". Received: {atlas}')
+
+    if parcellation is not None:
+        if not isinstance(data, np.ndarray):
```

Comment:
Just so I understand: what's the expected use-case here ? Someone provides a tuple of e.g., nifti image filepaths to the function along with a parcellation ? (Which wouldn't work, since we're expected pre-parcellated data when `parcellation is not None`.)

If that's the case then maybe you could just use `np.asarray(data, dtype='float32')`, such that when a tuple-of-pathlike objects is passed you'll get a `ValueError` which you could catch:

```
if parcellation is not None:
    try:
        data = np.asarray(data, dtype='float32')
    except ValueError as err:
        msg = ('`data` must be array_like when `parcellation` is not "None". '
               f'Received: {type(data)}')
        raise TypeError(msg) from err
```

(Note the `raise TypeError(msg) from err`—this makes it so you [don't get a terrible traceback](https://docs.python.org/3/tutorial/errors.html#exception-chaining) with "During handling of the above exception, another exception occurred")